### PR TITLE
MCOL-3606 Make ColumnStore use generic paths

### PR DESCRIPTION
**Pull request by LinuxJedi**
ColumnStore now uses standard bin/lib paths for pretty much everything.
Data path is now hard-coded to /var/lib/columnstore.

This patch also:

* Removes v1 decompression
* Removes a bunch of unneeded files
* Removes COLUMNSTORE_INSTALL_DIR / $INSTALLDIR
* Makes my.cnf.d work for all platforms (MCOL-3558)
* Changes configcpp to use recursive mutex (fixes possible config write deadlock)
* Fixes MCOL-3599 Fix regr functions, The library was installed in the wrong location
* Fixes a bunch of Ubuntu packaging issues
* Changes the binary names of several of the executables so as not to
clash with potential executables from other packages

### CODE REVIEW

**Comment by LinuxJedi** [comment]
Still some testing to do before this should be reviewed

**Comment by LinuxJedi** [comment]
Testing done, ready for review

File: oam/post/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mcstest-001.sh.in" "${CMAKE_CURRENT_SOURCE_DIR}/mcstest-001.sh" @ONLY)
```
Comment (drrtuy) [inline oam/post/CMakeLists.txt:1]: Do we really need these files?

Comment (LinuxJedi) [reply oam/post/CMakeLists.txt:1]: Apparently, yes. They appear to be used by ProcMon to make sure the node is really up.

ProcMon calls startupTests.sh which calls mcstest-*.sh

File: utils/configcpp/configcpp.h [hunk 2]
```diff
@@ -230,8 +230,7 @@ class Config
     xmlDocPtr fDoc;
     const std::string fConfigFile;
     time_t fMtime;
-    mutable boost::mutex fLock;
-    const std::string fInstallDir;
+    mutable boost::recursive_mutex fLock;
```
Comment (drrtuy) [inline utils/configcpp/configcpp.h:233]: Don't we want to make it std::recursive_mutex?

Comment (LinuxJedi) [reply utils/configcpp/configcpp.h:233]: No, because this is used for scoped locks which don't come into std until C++17

**Review by drrtuy — APPROVED** [review]

**Pull request merged**